### ISSUE TITLE
Tags filtering is broken

### ISSUE THREAD

**Issue by Luneyes** [kwaschny/unwanted-twitch#60]
Hi!
The tag filtering completely broke recently even though channels and categories one still work flawlessly.
It seems like Twitch changed their code again. :/

This extension is awesome, thank you for your hard work!

**Comment by kwaschny** [comment]
Are you using Chrome and the most recent UTTV version `21.5.22`? What exactly broke? Does it show the `X` buttons? Or does it just not remove items with blacklisted tags? What views are affected? Example links and screenshots help.

**Comment by Luneyes** [comment]
_Sorry for the lack of details, let me fix that!_

I'm using Firefox.
Version is `21.3.21` (up to date according to Firefox's store).

**What isn't working:**
- The `X` buttons don't show up on tags' names.
- Tags set up in the settings don't hide vids with such tags.
- Live Channels, Videos, Following, and Games views are affected.

**What works:**
- Tags filtering on Browse view.
- `X` button on tags, channels, and games in Browse view.
- Blacklisted names and game categories on every views.
- Manage blacklists button (and the eye button too).

Here is a screenshot on a Game view (and my settings) where tags filtering doesn't work and `X` button doesn't show up:
![Untitled](https://user-images.githubusercontent.com/26780000/119277181-684f4b80-bc1e-11eb-8438-3e49a9e825f2.png)

Here is a screenshot of Categories, everything work as intended:
![Untitled2](https://user-images.githubusercontent.com/26780000/119277490-55d61180-bc20-11eb-826f-d8aff1fa22f4.png)

**Comment by Luneyes** [comment]
Wait... something is wrong.
Apparently the tags `X` buttons sometimes work and sometimes don't...?
I tried refreshing some pages and they showed up, and on some other pages they disappeared. It seems quite random...
The channels' `X`'s and blacklists always work though.

I've got no other extensions that can interfere, I disabled all others just in case for the test.
Any ideas of what could cause this? Anything you want me to try?

**Comment by kwaschny** [comment]
> I'm using Firefox.
Version is 21.3.21 (up to date according to Firefox's store).

You are still using the old/outdated version. The most recent version is `21.5.22`. **Unfortunately all Firefox and Edge users will have to wait a few more days before the update is officially available via browser store, i. e. automatic update.** I'm sorry for making you write down everything in detail in this case. Let me tell you, this is still an excellent bug report. Thank you for your time. I added a quick solution at the bottom of this post, if you don't want to wait for the official store release.

For future reference, I will explain the flow in detail:

### tl;dr
**When I release a new version here on GitHub, it is not immediately available in the browser stores. And even once the stores serve the updated version, your browser still might not immediately update to the newest version.**

-----

This extension's code is shared between Chrome/Edge **and** Firefox. When I push a new version (see [commits](/kwaschny/unwanted-twitch/commits/master)), I publish the package via [Chrome Web Store](https://chrome.google.com/webstore), [Edge-Add-Ons](https://microsoftedge.microsoft.com/addons) (both package "chrome") and [Firefox Add-ons](https://addons.mozilla.org/) (package "firefox"). These stores have slightly different review policies to protect users against harmful software (and privacy violation). Firefox has historically been the fastest to automatically accept updates to existing extensions (usually less than 15 minutes), followed by Chrome (usually less than 3 hours) and Edge being the slowest (up to 10 days) due to manual/human review.

**However**, one of Mozilla's editorial managers (responsible for Firefox Add-ons) has recently informed me that they did a security code review, decided the extension is "trustworthy" (no warning about 3rd party extension on the store page) and will now require all new updates to be reviewed manually/by a human first. The reason is a marketing campaign to promote interesting extensions for Firefox. And they consider UTTV to be a candidate, which is really cool.

The downside is: It now takes even more time to react to changes on twitch.tv. It takes time to:
- realize something is broken
- file a bug report ([meme](https://i.imgur.com/3luZOx5.png) and [actual how to](/kwaschny/unwanted-twitch/blob/master/CONTRIBUTING.md))
- developer time to analyze the changes and adjust/fix the code
- wait for browser stores to approve the updated extension
- wait for user's browser to update to the newest version
- waiting for user feedback ([I have a test routine](/kwaschny/unwanted-twitch/blob/master/tests/acceptance-tests.md), but you miss the forest for the trees)
- repeat above steps until the majority of users is satisfied 🙂

-----

You can fast-forward the whole situation by installing the extension manually for the time being. It's pretty easy to do on Firefox:

- download the newest version of UTTV: [21.5.22](https://github.com/kwaschny/unwanted-twitch/raw/master/publish/firefox/21.5.22.zip?raw) (you can find future versions [here](https://github.com/kwaschny/unwanted-twitch/tree/master/publish/firefox))
- go to the Add-ons page in Firefox (`about:addons`)
- disable or remove the outdated Unwanted Twitch extension
- click on the cogwheel at the top and choose "Install addon from file"
- a file dialog appears where you select the just downloaded `21.5.22.zip` file
- done (note that the manually installed extension will never update itself, so I recommend switching back to the official one as soon as it is available)

**Comment by Luneyes** [comment]
Wow thank you for your precise answer, that's amazing!
Sadly Firefox disabled the possibility to add non-verified addons since v44 (even by editing stuff in the config page), so it's not possible anymore to install any addons that don't come from the store.
I guess I'll wait for the store to catch up. ¯\_(ツ)_/¯

PS: I'm stealing that bug report meme for later, it's very accurate. :D

**Comment by kwaschny** [comment]
> Firefox disabled the possibility to add non-verified addons [...]

Oh true, it only works with `about:debugging#/runtime/this-firefox` and then load it as temporary add-on. That's bit too inconvenient for daily use, I guess.

**Comment by Luneyes** [comment]
I can confirm it's fixed!
Thank you so much for your time and your help, what this extension does should be default features imo.
